BusinessLogic/Classes/TestProgressData.cs

```diff
@@ -1,20 +1,276 @@
+                {
+                    //not done testing with the selected ear ear
+                    //check if there is a last answer
+                    var answersOfSelectedEar = CurrentToneAudioMetryAnswers.Where(x => x.Item2 == CurrentEar).ToList();
+                    var lastAnswer = answersOfSelectedEar[answersOfSelectedEar.Count() - 1];
+                    Decibel = lastAnswer.Item4 + (int)returnedDecibel;
+                }
+                else if (returnedDecibel == null)
+                {
+                    //done with testing of selected ear so moving onto the other ear
+                    CurrentEar = (CurrentEar == Ear.Left) ? Ear.Right : Ear.Left;
+                    returnedDecibel = GetNextDecibels(CurrentEar);
+                    if (returnedDecibel != null)
+                    {
+                        //check if there is a last answer
+                        var answersOfSelectedEar = CurrentToneAudioMetryAnswers.Where(x => x.Item2 == CurrentEar).ToList();
+                        var lastAnswer = answersOfSelectedEar[answersOfSelectedEar.Count() - 1];
+                        Decibel = lastAnswer.Item4 + (int)returnedDecibel;
+                    }
+                }
+            }
+
+            return Test.ToneAudiometryQuestions.First(x => x.QuestionNumber == CurrentQuestionNumber);
+        }
+    }
+
+    private bool IsFrequencyDone()
+    {
+        //if the list.count() <= 0 for either ear then there is no use to call GetNextDecibel.
+        //It means the test hasnt even begun yet and therefore IsFrequencyDone is false.
+        var leftEarAnswer = CurrentToneAudioMetryAnswers.Where(x => x.Item2 == Ear.Left).ToList();
+        var rightEarAnswer = CurrentToneAudioMetryAnswers.Where(x => x.Item2 == Ear.Right).ToList();
+        int? resultLeft;
+        int? resultRight;
+        
+        if (leftEarAnswer.Count() > 0)
+        {
+            resultLeft = GetNextDecibels(Ear.Left);
+        }
+        else
+        {
+            return false;
+        }
+
+        if (rightEarAnswer.Count() > 0)
+        {
+            resultRight = GetNextDecibels(Ear.Right);
+        }
+        else
+        {
+            return false;
+        }
+
+        return resultLeft == null && resultRight == null;
+    }
+
+    private int? GetNextDecibels(Ear ear)
+    {
+        var answersOfSelectedEar = CurrentToneAudioMetryAnswers.Where(x => x.ear == ear).ToList();
+        var lastAnswer = answersOfSelectedEar[answersOfSelectedEar.Count() - 1];
+        //last answer was true
+        if (lastAnswer.answer)
+        {
+            bool falseGivenAnswer = false;
+            //check if any of the previous answers were false (opposite of the last given answer)
+            foreach (var item in answersOfSelectedEar)
+            {
+                if (item.answer != lastAnswer.answer)
+                {
+                    falseGivenAnswer = true;
+                    break;
+                }
+            }
```

Comment:
Misschien is dit wat simpeler (zelfde geldt voor het andere antwoord)

```suggestion
            bool hasGivenFalseAnswer = answersOfSelectedEar.Where(item => item.answer == false).Count() > 0;
```

Reply:
je kan zelfs doen   bool hasGivenFalseAnswer = answersOfSelectedEar.Where(item => !item.answer).Count() > 0;